src/App.test.tsx

```diff
@@ -1,49 +1,268 @@
-  expect(screen.getByText('Load Contract')).toBeInTheDocument()
-})
+  afterEach(() => {
+    // Running all pending timers and switching to real timers using Jest
+    jest.runOnlyPendingTimers()
+    jest.useRealTimers()
+    jest.resetModules()
+  })
 
-test('navigates to Administrate', () => {
-  render(<App />)
-  userEvent.click(screen.getByText('Administrate'))
-  expect(screen.getByText('Allocate Tokens')).toBeInTheDocument()
-})
+  describe('open the create page and', () => {
+    beforeEach(() => {
+      render(
+        <MemoryRouter>
+          <App />
+        </MemoryRouter>
+      )
+    })
 
-test('navigates to settings', () => {
-  render(<App />)
-  userEvent.click(screen.getByText('SettingsPage'))
-  expect(screen.getByText('Wallet SettingsPage')).toBeInTheDocument()
-})
+    it('render the page', () => {
+      const elements = ['Create', 'Vote', 'Administrate', 'Settings', 'Unlock Wallet']
+      elements.forEach((text) => {
+        const element = screen.getByText(text)
+        expect(element).toBeInTheDocument()
+      })
+    })
+
+    it('navigate correctly', () => {
+      fireEvent.click(screen.getByText('Vote'))
+      expect(screen.getByText('Load Contract')).toBeInTheDocument()
+      fireEvent.click(screen.getByText('Administrate'))
+      expect(screen.getByText('Allocate Tokens')).toBeInTheDocument()
+      fireEvent.click(screen.getByText('Settings'))
+      expect(screen.getByText('Wallet SettingsPage')).toBeInTheDocument()
+    })
+
+    it('unlock the wallet', () => {
+      fireEvent.click(screen.getByRole('button', { name: 'Unlock Wallet' }))
+      expect(Client.prototype.walletUnlock).toHaveBeenCalledTimes(1)
+    })
+
+    it('deploy the voting contract', async () => {
+      const adminInput = screen.getByLabelText('Administrator Address')
+      const voterInput = screen.getByPlaceholderText('Enter voter address')
+      const addVoterBtn = screen.getByRole('button', { name: '+' })
+      const submitBtn = screen.getByRole('button', { name: 'Submit' })
+
+      fireEvent.click(screen.getByRole('button', { name: 'Unlock Wallet' }))
+      await waitFor(() => expect(Client.prototype.walletUnlock).toHaveBeenCalledTimes(1))
+      fireEvent.change(adminInput, { target: { value: adminAddress } })
+      voters.forEach((voter) => {
+        fireEvent.change(voterInput, { target: { value: voter } })
+        fireEvent.click(addVoterBtn)
+      })
+      fireEvent.click(submitBtn)
+      await waitFor(() => {
+        expect(Client.prototype.contractSubmissionPipeline).toHaveBeenCalledWith(
+          createContract(voters.length),
+          CONTRACTGAS,
+          initContractState(adminAddress, voters),
+          voters.length.toString()
+        )
+        expect(Client.prototype.contractSubmissionPipeline).toHaveBeenCalledTimes(1)
+        expect(Client.prototype.getTxStatus).toHaveBeenCalledTimes(1)
+        expect(screen.getByText('confirmed !')).toBeInTheDocument()
```

Comment:
Is adding a space before punctuation a French thing? 😅

Reply:
Yes it is 😅, will fix this